api/v1alpha3/awscluster_types.go

```diff
@@ -82,6 +82,11 @@ type AWSClusterSpec struct {
 	// Bastion contains options to configure the bastion host.
 	// +optional
 	Bastion Bastion `json:"bastion"`
+
+	// SecretBackend is an optional string allowing the user to specify either Secrets Manger or SSM Parameter Storage.
+	// Only possible input is 'ssm', any other entry will default to Secrets Manager.
+	// +optional
+	SecretBackend string `json:"secretBackend,omitempty`
```

Comment:
How about creating a type for the secret backend? And then having 2 instances of that type for the secrets manager and parameters store? Like we do for the [ELB scheme](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/api/v1alpha3/types.go#L82). The values for the 2 types would need to be the same as the enum values in the validation above.